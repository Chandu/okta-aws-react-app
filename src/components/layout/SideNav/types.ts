export interface Props {
    activeSideBarPath?: string;
    defaultOpen?: boolean;
    handlePathChange?: (page: string) => void;
}

export interface SideBarMenuConf {
    path: string;
    icon: JSX.Element;
}
