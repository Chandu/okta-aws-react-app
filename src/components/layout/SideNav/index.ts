export { AppSideNav as default } from './SideNav';
