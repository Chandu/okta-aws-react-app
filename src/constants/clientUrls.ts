export const URLRoot = {
    dashboard: '/',
    dataUpload: '/data-upload'
};

export const AuthURL = {
    login: '/login'
};
