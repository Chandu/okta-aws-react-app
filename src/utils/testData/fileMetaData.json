{
    "description": "dummy description",
    "uploaded-by-user-name": "xyz",
    "uploaded-timestamp": "Fri Apr 29 2022 18:46:10 GMT+0530 (India Standard Time)",
    "uploaded-user-email": "xyz@gmail.com"
}
