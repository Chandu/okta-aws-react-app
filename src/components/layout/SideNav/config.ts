import { AuthURL, URLRoot as URL } from '@constants/clientUrls';
import { matchPath } from 'react-router-dom';

const sideNavConfig = {
    [URL.dashboard]: { title: 'Dashboard' },
    [URL.dataUpload]: { title: 'Data upload' }
};

export const getNavItemTitle = (path: string) => sideNavConfig[path] && sideNavConfig[path].title;

export const pathsWhereSidebarHidden = [...Object.values(AuthURL)];

export const isSidebarVisible = (pathname: string) =>
    pathsWhereSidebarHidden.every(
        path =>
            matchPath(pathname, {
                path,
                exact: true,
                strict: false
            }) === null
    );
