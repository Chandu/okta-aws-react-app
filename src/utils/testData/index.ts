import fileMetaData from './fileMetaData.json';
import fileUploadHistory from './fileUploadHistory.json';
import oktaStorage from './oktaStorage.json';

export { oktaStorage, fileMetaData, fileUploadHistory };
