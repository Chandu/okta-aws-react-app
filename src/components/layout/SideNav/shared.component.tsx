import { Text } from '@medly-components/core';
import { SideNav as MedlySideNav } from '@medly-components/layout';
import { getNavItemTitle } from './config';
import { SideBarMenuConf } from './types';

export const getMedlySideBarNav = (menus: SideBarMenuConf[]) =>
    menus.map(conf => (
        <MedlySideNav.Nav key={conf.path} path={conf.path}>
            {conf.icon}
            <Text>{getNavItemTitle(conf.path)}</Text>
        </MedlySideNav.Nav>
    ));
