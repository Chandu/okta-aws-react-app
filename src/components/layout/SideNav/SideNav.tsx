import { URLRoot as URL } from '@constants/clientUrls';
import { UploadIcon } from '@icons';
import { InsertChartOutlinedIcon } from '@medly-components/icons';
import { SideNav as MedlySideNav } from '@medly-components/layout';
import { WithStyle } from '@medly-components/utils';
import { FC, memo, useCallback } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { isSidebarVisible } from './config';
import { getMedlySideBarNav } from './shared.component';
import { SideNavHeader } from './SidenavHeader/SideNavHeader';
import { Props, SideBarMenuConf } from './types';

const menuList: SideBarMenuConf[] = [
    {
        path: '/',
        icon: <InsertChartOutlinedIcon />
    },
    {
        path: URL.dataUpload,
        icon: <UploadIcon />
    }
];

const Component: FC<Props> = memo(props => {
    const { pathname } = useLocation(),
        sideBarVisible = isSidebarVisible(pathname),
        { defaultOpen } = props,
        history = useHistory(),
        handlePathChange = useCallback((page: string) => history.push(page), [history]);
    if (!sideBarVisible) return null;

    return (
        <MedlySideNav
            onChange={handlePathChange}
            active={pathname}
            defaultOpen={defaultOpen}
            expandedToggleText="Hide menu"
            collapsedToggleText="Open menu"
        >
            <SideNavHeader />
            <MedlySideNav.List>
                <MedlySideNav.Group title="Menu">{getMedlySideBarNav(menuList)}</MedlySideNav.Group>
            </MedlySideNav.List>
        </MedlySideNav>
    );
});

Component.displayName = 'AppSideNav';
Component.defaultProps = {
    defaultOpen: true
};
export const AppSideNav: FC<Props> & WithStyle = Object.assign(Component, {
    Style: MedlySideNav.Style
});
