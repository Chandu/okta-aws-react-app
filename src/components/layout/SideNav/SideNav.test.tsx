import { URLRoot } from '@constants/clientUrls';
import { renderWithRouter, screen } from '@test-utils';
import { oktaStorage } from '@utils/testData';
import { MemoryRouter } from 'react-router-dom';
import { pathsWhereSidebarHidden } from './config';
import { AppSideNav } from './SideNav';

const mockHistoryPush = jest.fn();
jest.mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useHistory: () => ({
        push: mockHistoryPush
    })
}));

const renderSideNav = (url: string = URLRoot.dashboard) =>
    renderWithRouter(
        <MemoryRouter initialEntries={[url]}>
            <AppSideNav />
        </MemoryRouter>
    );

describe('SideNav', () => {
    it('should render properly', () => {
        localStorage.setItem('okta-token-storage', JSON.stringify(oktaStorage));
        const { container } = renderSideNav(URLRoot.dashboard);
        expect(screen.getByText('Dashboard')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('should not render sidebar where it is hidden', () => {
        localStorage.setItem('okta-token-storage', JSON.stringify(oktaStorage));
        const { container } = renderSideNav(pathsWhereSidebarHidden[0]);
        expect(screen.queryByText('Dashboard')).not.toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('should not render sidebar when user is not authenticated', () => {
        renderSideNav(pathsWhereSidebarHidden[0]);
        expect(screen.queryByText('Dashboard')).not.toBeInTheDocument();
    });
});
