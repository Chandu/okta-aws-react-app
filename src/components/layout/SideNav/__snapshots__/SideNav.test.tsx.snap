// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav should not render sidebar where it is hidden 1`] = `<div />`;

exports[`SideNav should render properly 1`] = `
.c5 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
  padding: 1rem;
  border-radius: 0.4rem;
  background-color: transparent;
}

.c5 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #6E14FF;
}

.c19 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c19 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c10 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 500;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  text-transform: uppercase;
  display: inline-block;
}

.c17 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 500;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 2rem;
  text-align: initial;
  text-transform: uppercase;
}

.c20 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  text-align: initial;
  display: inline-block;
}

.c23 {
  margin: 0;
  color: inherit;
  font-size: 1.2rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 1.6rem;
  text-align: initial;
  display: inline-block;
}

.c7 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 {
  height: 100vh;
  height: -webkit-fill-available;
  height: -moz-available;
  height: fill-available;
  background-color: #dfe4e9;
  width: 0;
  position: fixed;
  z-index: 10;
  top: 0;
  -webkit-transition: 0.2s;
  transition: 0.2s;
  left: 0;
}

.c14 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0;
  margin: 0;
  list-style-type: none;
  overflow-x: hidden;
  overflow-y: auto;
}

.c14::-webkit-scrollbar {
  width: 0.5rem;
  background-color: #F7F8F9;
}

.c14::-webkit-scrollbar-thumb {
  border-radius: 1rem;
  background-color: #98A7B7;
}

.c16 {
  position: relative;
}

.c16 > .c9 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  margin: 3.1rem 2.4rem 1.4rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: block;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 1;
  color: #98A7B7;
}

.c16 .c13 {
  overflow: visible;
}

.c16:not(:first-child)::before {
  content: '';
  width: 0;
  -webkit-transition: width 200ms ease-out;
  transition: width 200ms ease-out;
  margin: 0 auto;
  display: block;
  position: absolute;
  border-top: 0 solid #dfe4e9;
  top: 2rem;
  left: 0;
  right: 0;
}

.c18 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  margin: 0 1.2rem;
  border-radius: 0.4rem;
  box-shadow: 0 0 2.4rem #eff2f4;
  grid-template-columns: calc(7.2rem - 1.2rem) calc(25.6rem - 7.2rem - 1.2rem);
  justify-items: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: pointer;
  background-color: #ffffff;
}

.c18 .c4 {
  overflow: visible;
  border-radius: 0.4rem;
  padding: calc((4.8rem - 2rem) / 2);
  font-size: 2rem;
}

.c18 .c9 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  overflow: hidden;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 1;
  font-size: 1.4rem;
  font-weight: 500;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  font-family: Poppins,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c18 .c4 * {
  fill: #6E14FF;
}

.c18 .c9 {
  color: #13181D;
}

.c21 {
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  display: grid;
  margin: 0 1.2rem;
  border-radius: 0.4rem;
  grid-template-columns: calc(7.2rem - 1.2rem) calc(25.6rem - 7.2rem - 1.2rem);
  justify-items: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  min-height: 4.8rem;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  cursor: pointer;
  background-color: transparent;
}

.c21 .c4 {
  overflow: visible;
  border-radius: 0.4rem;
  padding: calc((4.8rem - 2rem) / 2);
  font-size: 2rem;
}

.c21 .c9 {
  text-overflow: ellipsis;
  white-space: nowrap;
  justify-self: left;
  overflow: hidden;
  -webkit-transition: opacity 200ms ease-out,color 100ms ease-out;
  transition: opacity 200ms ease-out,color 100ms ease-out;
  opacity: 1;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2rem;
  font-family: Poppins,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c21 .c4 * {
  fill: #607890;
}

.c21 .c9 {
  color: #435465;
}

.c21:hover {
  background-color: #eff2f4;
}

.c21:hover .c4 * {
  fill: #303C48;
}

.c21:hover .c9 {
  color: #303C48;
}

.c21:active {
  background-color: #dfe4e9;
}

.c21:active .c4 * {
  fill: #13181D;
}

.c21:active .c9 {
  color: #13181D;
}

.c1 {
  position: inherit;
  -webkit-transition: width 200ms ease-out;
  transition: width 200ms ease-out;
  width: 25.6rem;
  box-shadow: 0.2rem 0 8px rgba(96,120,144,0.2);
}

.c2 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: hidden;
  -webkit-transition: all 200ms ease-out;
  transition: all 200ms ease-out;
  background-color: #F7F8F9;
  width: 25.6rem;
  box-shadow: none;
}

.c2 > .c13:first-of-type {
  padding-top: 3.7rem;
}

.c2 > .c13:first-of-type > .c15:first-of-type > .c9 {
  margin-top: 0.6rem;
}

.c24 {
  min-height: 7.2rem;
  -webkit-flex: unset;
  -ms-flex: unset;
  flex: unset;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
  position: relative;
  width: 100%;
  margin-top: 2rem;
}

.c24::before {
  content: '';
  width: calc(100% - 4.0rem);
  margin: 0 auto;
  display: block;
  position: absolute;
  border-top: 1px solid #dfe4e9;
  top: 0;
  left: 0;
  right: 0;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  min-width: -webkit-fit-content;
  min-width: -moz-fit-content;
  min-width: fit-content;
  opacity: 1;
  -webkit-transition: opacity 200ms ease-out;
  transition: opacity 200ms ease-out;
  white-space: nowrap;
  margin: 1.6rem 2.4rem 0.6rem;
  color: #546A7F;
}

.c22 span {
  margin-right: 0.2rem;
}

.c22 img {
  margin-top: 0.1rem;
  margin-left: 0.8rem;
}

.c11 {
  margin-right: 1.4rem;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  text-transform: capitalize;
}

.c6 {
  z-index: inherit;
  border: 0.1px solid #C7D0D8;
  padding: 0.9rem;
  background-color: transparent;
}

.c6 > path {
  fill: #6E14FF;
}

.c3 {
  width: 100%;
  z-index: 3;
  min-height: 7.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
  display: grid;
  overflow: visible;
  grid-template-columns: 7.2rem calc(25.6rem - 7.2rem);
  color: #435465;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c12 {
  border-bottom: 0.1rem solid #dfe4e9;
  width: calc(100% - 4rem);
  margin: auto;
  margin-top: -0.1rem;
}

.c8 {
  position: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c8 .c9 {
  cursor: default;
}

<div>
  <aside
    class="c0 c1"
    id="medly-sidenav"
  >
    <nav
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          class="c4 c5 c6"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12 5v2h8c1.1 0 2 .9 2 2v10c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h6c1.1 0 2 .9 2 2zM4 19h2v-2H4v2zm2-4H4v-2h2v2zm-2-4h2V9H4v2zm2-4H4V5h2v2zm2 12h2v-2H8v2zm2-4H8v-2h2v2zm-2-4h2V9H8v2zm2-4H8V5h2v2zm2 12h7c.55 0 1-.45 1-1v-8c0-.55-.45-1-1-1h-7v2h2v2h-2v2h2v2h-2v2zm6-8h-2v2h2v-2zm-2 4h2v2h-2v-2z"
            fill="#000"
            fill-opacity=".54"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="c7 c8"
          id="medly-popover-wrapper"
        >
          <span
            class="c9 c10 c11"
          >
            Medicare
          </span>
        </div>
      </div>
      <div
        class="c12"
      />
      <ul
        class="c13 c14"
      >
        <li
          class="c15 c16"
        >
          <h5
            class="c9 c17"
          >
            Menu
          </h5>
          <ul
            class="c13 c14"
          >
            <li
              class="c18"
            >
              <svg
                class="c4 c19"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M5 3h14c1.1 0 2 .9 2 2v14c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2zm1 16c-.55 0-1-.45-1-1V6c0-.55.45-1 1-1h12c.55 0 1 .45 1 1v12c0 .55-.45 1-1 1H6zm6-2c-.55 0-1-.45-1-1V8c0-.55.45-1 1-1s1 .45 1 1v8c0 .55-.45 1-1 1zm-4 0c-.55 0-1-.45-1-1v-5c0-.55.45-1 1-1s1 .45 1 1v5c0 .55-.45 1-1 1zm7-1c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1v2z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c9 c20"
              >
                Dashboard
              </span>
            </li>
            <li
              class="c21"
            >
              <svg
                class="c4 c19"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 13H4c-.55 0-1-.45-1-1V4c0-.55.45-1 1-1h6c.55 0 1 .45 1 1v8c0 .55-.45 1-1 1zm0 8H4c-.55 0-1-.45-1-1v-4c0-.55.45-1 1-1h6c.55 0 1 .45 1 1v4c0 .55-.45 1-1 1zm4 0h6c.55 0 1-.45 1-1v-8c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1zM13 8V4c0-.55.45-1 1-1h6c.55 0 1 .45 1 1v4c0 .55-.45 1-1 1h-6c-.55 0-1-.45-1-1z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c9 c20"
              >
                Reports
              </span>
            </li>
            <li
              class="c21"
            >
              <svg
                class="c4 c19"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M15.5 1h-11c-1.1 0-2 .9-2 2v13c0 .55.45 1 1 1s1-.45 1-1V4c0-.55.45-1 1-1h10c.55 0 1-.45 1-1s-.45-1-1-1zm4 4h-11c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm-10 16h9c.55 0 1-.45 1-1V8c0-.55-.45-1-1-1h-9c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c9 c20"
              >
                Contract summary
              </span>
            </li>
            <li
              class="c21"
            >
              <svg
                class="c4 c19"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M20.99 2.21H21h-.01zm0 0a.468.468 0 0 0-.34.15l-.79.79c-.2.2-.51.2-.71 0l-.79-.79c-.2-.2-.51-.2-.71 0l-.79.79c-.2.2-.51.2-.71 0l-.79-.79c-.2-.2-.51-.2-.71 0l-.79.79c-.2.2-.51.2-.71 0l-.79-.79c-.2-.2-.51-.2-.71 0l-.79.79c-.2.2-.51.2-.71 0l-.8-.8c-.2-.2-.51-.2-.71 0l-.79.8c-.2.2-.51.2-.71 0l-.79-.8c-.2-.2-.51-.2-.71 0l-.79.8c-.2.2-.51.2-.71 0l-.79-.8A.5.5 0 0 0 3 2.21V21.8c.13 0 .26-.05.35-.15l.79-.79c.2-.2.51-.2.71 0l.79.79c.2.2.51.2.71 0l.79-.79c.2-.2.51-.2.71 0l.79.79c.2.2.51.2.71 0l.79-.79c.2-.2.51-.2.71 0l.79.79c.2.2.51.2.71 0l.79-.79c.2-.2.51-.2.71 0l.79.79c.2.2.51.2.71 0l.79-.79c.2-.2.51-.2.71 0l.79.79c.2.2.51.2.71 0l.79-.79c.2-.2.51-.2.71 0l.79.79c.1.1.23.15.35.15V2.21zM7 17h10c.55 0 1-.45 1-1s-.45-1-1-1H7c-.55 0-1 .45-1 1s.45 1 1 1zm10-4H7c-.55 0-1-.45-1-1s.45-1 1-1h10c.55 0 1 .45 1 1s-.45 1-1 1zM7 9h10c.55 0 1-.45 1-1s-.45-1-1-1H7c-.55 0-1 .45-1 1s.45 1 1 1z"
                  fill="#000"
                  fill-opacity=".54"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="c9 c20"
              >
                Invoice
              </span>
            </li>
          </ul>
        </li>
      </ul>
      <div
        class="c22"
      >
        <span
          class="c9 c23"
        >
          Powered by
        </span>
        <img
          alt="Khora logo"
          src="/images/secondary-logo.svg"
        />
      </div>
      <ul
        class="c13 c14 c24"
      >
        <li
          class="c21"
        >
          <svg
            class="c4 c19"
            fill="none"
            height="1em"
            id="medly-sidenav-toggle-hide"
            title="medly-sidenav-toggle-hide"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.889 18H22.11c.489 0 .889-.45.889-1s-.4-1-.889-1H7.89C7.4 16 7 16.45 7 17s.4 1 .889 1zM7.889 13H22.11c.489 0 .889-.45.889-1s-.4-1-.889-1H7.89C7.4 11 7 11.45 7 12s.4 1 .889 1zM7 7c0 .55.4 1 .889 1H22.11C22.6 8 23 7.55 23 7s-.4-1-.889-1H7.89C7.4 6 7 6.45 7 7zM3.824 8.789l-2.537 2.525a.967.967 0 0 0 0 1.374l2.537 2.525c.618.615 1.676.176 1.676-.692v-5.05c0-.867-1.058-1.296-1.676-.682z"
              fill="#000"
            />
          </svg>
          <span
            class="c9 c20"
          >
            Hide menu
          </span>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
